### PR TITLE
Add GammaTest2 to RVSDG tests

### PR DESCRIPTION
**Pull request by phate**
The PR title says it all.

### CODE REVIEW

**Comment by phate** [comment]
@sjalander @caleridas @haved : Could I get a review, please?

File: tests/TestRvsdgs.cpp [hunk 1]
```diff
@@ -1288,6 +1288,205 @@ GammaTest::SetupRvsdg()
...
+
+      // gamma subregion 0
+      auto loadXResults = LoadNode::Create(
+        gammaInputX->argument(0),
+        {gammaInputMemoryState->argument(0)},
+        jlm::rvsdg::bit32,
+        4);
+
+      auto one = rvsdg::create_bitconstant(gammaNode->subregion(0), 32, 1);
+      auto storeZRegion0Results = StoreNode::Create(
```
Comment (sjalander) [inline tests/TestRvsdgs.cpp:1326]: Why does the store node not have a memory state?

File: tests/TestRvsdgs.cpp [hunk 2]
```diff
@@ -1288,6 +1288,205 @@ GammaTest::SetupRvsdg()
...
+      // gamma subregion 1
+      auto loadYResults = LoadNode::Create(
+        gammaInputY->argument(1),
+        {gammaInputMemoryState->argument(1)},
+        jlm::rvsdg::bit32,
+        4);
+
+      auto two = rvsdg::create_bitconstant(gammaNode->subregion(1), 32, 2);
+      auto storeZRegion1Results = StoreNode::Create(
+        gammaInputZ->argument(1),
```
Comment (sjalander) [inline tests/TestRvsdgs.cpp:1341]: Same here

File: tests/TestRvsdgs.hpp [hunk 3]
```diff
@@ -855,6 +855,141 @@ class GammaTest final : public RvsdgTest {
 	jlm::rvsdg::gamma_node * gamma;
 };
 
+/** \brief GammaTest2 class
+ *
+ * This function sets up an RVSDG representing the following function:
```
Comment (sjalander) [inline tests/TestRvsdgs.hpp:860]: It's a class and no specific function is mentioned.

File: tests/TestRvsdgs.hpp [hunk 4]
```diff
@@ -855,6 +855,141 @@ class GammaTest final : public RvsdgTest {
...
+ *
+ * \code{.c}
```
Comment (sjalander) [inline tests/TestRvsdgs.hpp:862]: Not fluent enough to read construction of RVSDG to tell if it is a true representation of the this C code

**Review by sjalander — APPROVED** [review]

File: tests/TestRvsdgs.cpp (hunk 1 above)
Comment (phate) [reply tests/TestRvsdgs.cpp:1326]: It has a memory state: `{loadXResults[1]},`

File: tests/TestRvsdgs.cpp (hunk 2 above)
Comment (phate) [reply tests/TestRvsdgs.cpp:1341]: Same here: `{loadYResults[1]},`

File: tests/TestRvsdgs.hpp (hunk 3 above)
Comment (phate) [reply tests/TestRvsdgs.hpp]: Fixed.

**Pull request merged**